### PR TITLE
Updating default finality tag settings 

### PR DESCRIPTION
**Pull request by mateusz-sekara**
The default value for all chains was false. Enabling finality tags for some chains that actually use it.

Followup to https://github.com/smartcontractkit/chainlink/pull/9607

### CODE REVIEW

File: core/chains/evm/config/toml/defaults/Avalanche_Fuji.toml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ChainID = '43113'
 FinalityDepth = 1
+FinalityTagEnabled = false
```
Comment (RensR) [inline core/chains/evm/config/toml/defaults/Avalanche_Fuji.toml:3]: let's add a comment why we set it to false even though technically it supports it, same for other Avax

**Review by dimriou — COMMENTED** [review]
Why are we enabling this on the main repo instead of CCIP? This config is to enable a future feature, not to see if the network supports the tag. The feature is yet unimplemented on the main repo, so I think it might confuse people as to why it is enabled here.

**Comment by mateusz-sekara** [comment]
> Why are we enabling this on the main repo instead of CCIP. This config is to enable a future feature, not to see if the network supports the tag. The feature is yet unimplemented on the main repo, so I think it might confuse people as to why it is enabled here.

We already have it in CCIP; therefore, I wanted to backport this change to the main repo. Based on the options present there, I thought this config describes chain/network configuration, not a list of enabled features per chain. Not sure about having different values for CCIP and Core, at some point, it might lead to some drift or bugs that are hard to spot. @connorwstein any suggestions?

File: core/chains/evm/config/toml/defaults/Avalanche_Mainnet.toml [hunk 2]
```diff
@@ -1,5 +1,9 @@
 ChainID = '43114'
 FinalityDepth = 1
+# eth_getBlockByNumber(latest) == eth_getBlockByNumber(finalized) https://docs.avax.network/build/dapp/advanced/integrate-exchange#determining-finality
+# Avalanche RPCs frequently experience rewinds, ie. they might report a block as finalized, and afterwards report a block at a lower height as finalized.
+# This violates an invariant that subsystems might expect, that finalized blocks reported always build on top of earlier reported finalized blocks.
```
Comment (dimriou) [inline core/chains/evm/config/toml/defaults/Avalanche_Mainnet.toml:5]: That's a very useful comment. Thanks for adding.

**Comment by connorwstein** [comment]
> > Why are we enabling this on the main repo instead of CCIP. This config is to enable a future feature, not to see if the network supports the tag. The feature is yet unimplemented on the main repo, so I think it might confuse people as to why it is enabled here.
> 
> We already have it in CCIP; therefore, I wanted to backport this change to the main repo. Based on the options present there, I thought this config describes chain/network configuration, not a list of enabled features per chain. Not sure about having different values for CCIP and Core, at some point, it might lead to some drift or bugs that are hard to spot. @connorwstein any suggestions?

Yeah I misunderstood - I thought it was config describing the network. As a feature flag then I think it makes sense to keep it for CCIP only, since CCIP supports it. Once its supported in core then we can enable it here

**Pull request closed without merging**